addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveTimeParametersCommandClass.java

```diff
@@ -0,0 +1,208 @@
+
+        ByteArrayOutputStream outputData = new ByteArrayOutputStream();
+        outputData.write(getNode().getNodeId());
+        outputData.write(9);
+        outputData.write(getCommandClass().getKey());
+        outputData.write(TIME_SET);
+        outputData.write((cal.get(Calendar.YEAR) & 0xff00) >> 8);
+        outputData.write((cal.get(Calendar.YEAR) & 0xff));
+        outputData.write(cal.get(Calendar.MONTH) + 1);
+        outputData.write(cal.get(Calendar.DAY_OF_MONTH));
+        outputData.write(cal.get(Calendar.HOUR_OF_DAY));
+        outputData.write(cal.get(Calendar.MINUTE));
+        outputData.write(cal.get(Calendar.SECOND));
+        result.setMessagePayload(outputData.toByteArray());
+        return result;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @throws ZWaveSerialMessageException
+     */
+    @Override
+    public void handleApplicationCommandRequest(SerialMessage serialMessage, int offset, int endpoint)
+            throws ZWaveSerialMessageException {
+        logger.debug("NODE {}: Received clock command (v{})", this.getNode().getNodeId(), this.getVersion());
+        int command = serialMessage.getMessagePayloadByte(offset);
+        switch (command) {
+            case TIME_GET:
+                logger.debug("NODE %d: requested the time", getNode().getNodeId());
+                getController().sendData(getSetMessage(new Date()));
+
+            case TIME_REPORT:
+
+                int year = (serialMessage.getMessagePayloadByte(offset + 1) << 8
+                        | serialMessage.getMessagePayloadByte(offset + 2));
+                int month = serialMessage.getMessagePayloadByte(offset + 3);
+                int day = serialMessage.getMessagePayloadByte(offset + 4);
+                int hour = serialMessage.getMessagePayloadByte(offset + 5);
+                int minute = serialMessage.getMessagePayloadByte(offset + 6);
+                int second = serialMessage.getMessagePayloadByte(offset + 7);
+
+                Calendar cal = Calendar.getInstance();
+                cal.clear();
+                cal.set(year, month - 1, day, hour, minute, second);
+                logger.debug("NODE {}: Received time report: {}", getNode().getNodeId(), cal.getTime());
+
+                Date nodeTime = cal.getTime();
+                long clockOffset = (nodeTime.getTime() - System.currentTimeMillis()) / 1000;
+                ZWaveTimeValueEvent zEvent = new ZWaveTimeValueEvent(getNode().getNodeId(), endpoint, clockOffset);
+                this.getController().notifyEventListeners(zEvent);
+
+                Date low = new Date(System.currentTimeMillis() - CLOCK_PRESISION);
+                Date high = new Date(System.currentTimeMillis() + CLOCK_PRESISION);
+
+                // auto correct the time if off by more than 1 minutes
+                if (nodeTime.before(low) || nodeTime.after(high)) {
+                    updateClock = true;
```

Comment:
Same again - I'd suggest to remove this and do it in the application where we have more flexibility